### PR TITLE
chore: commitizen ci actions

### PR DESCRIPTION
**Pull request by jarvis8x7b**

### CODE REVIEW

File: .github/workflows/commit-message.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: "Lint PR Commit Messages"
+
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review, synchronize, edited]
```
Comment (codebender37) [inline .github/workflows/commit-message.yaml]: I think we can remove the `edited` here since it doesn't need to trigger again when PR title is edited

Comment (jarvis8x7b) [reply .github/workflows/commit-message.yaml]: oh yeah, it is actually there in the case someone creates PR title by default (e.g. fix/wandb which fails) then the edit will trigger the job again right after

File: .github/workflows/commit-message.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          echo "$PWD/.venv/bin" >> $GITHUB_PATH
+
+      - name: Install dependencies
+        run: uv pip install commitizen
+
+      - name: Check commit messages
+        run: |
+          cz check \
+          --rev-range ${{ github.event.pull_request.base.sha }}..${{ github.event.pull_request.head.sha }} \
+          --allowed-prefixes 'Merge' 'Revert' 'Pull request' 'fixup!' 'squash!'
```
Comment (karootplx) [inline .github/workflows/commit-message.yaml:38]: what about other conventional commit types such as fix, feat etc

File: .github/workflows/pr_title.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+name: "Lint PR Title"
+
+on:
+  pull_request:
+    types: [opened, reopened, ready_for_review, synchronize, edited]
+    branches:
+      - dev
+      - staging
```
Comment (karootplx) [inline .github/workflows/pr_title.yaml]: no need staging

File: .github/workflows/commit-message.yaml (hunk 1 above)
Comment (codebender37) [reply .github/workflows/commit-message.yaml]: I see, but doesn't need to run on `commit-message.yaml` right? only on `pr_title.yaml` imo.

Comment (jarvis8x7b) [reply .github/workflows/commit-message.yaml]: oh yes, didn't see the filename

File: .github/workflows/commit-message.yaml (hunk 2 above)
Comment (jarvis8x7b) [reply .github/workflows/commit-message.yaml:38]: it's included by default, do have a read here https://commitizen-tools.github.io/commitizen/commands/check/#allowed-prefixes . basically any prefix here is for non-conventional commit prefixes

File: .github/workflows/pr_title.yaml (hunk 3 above)
Comment (jarvis8x7b) [reply .github/workflows/pr_title.yaml]: removed

File: .github/workflows/commit-message.yaml (hunk 1 above)
Comment (jarvis8x7b) [reply .github/workflows/commit-message.yaml]: removed

**Review by karootplx — APPROVED** [review]
lgtm

**Pull request merged**